app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class RepositoryTypeManager
```

Comment:
hrmmm, so actually while i prefer the implementation of the bastion plugin, it is must simpler taking a simple hash.  Using a dsl like language may be more advantageous in the end (or may not?) as instead of parsing the hash you're defining methods that are called at runtime to configure things.  For a large app plugin api i think that does make more sense that a simpler api taking a hash.  For this api I think it will eventually get more complicated that it is in this PR you could probably argue either way